### ISSUE TITLE
False positive unused-import (W0611) when assigning class properties with the same name

### ISSUE THREAD

**Issue by teake** [PyCQA/pylint#6089]
### Bug description

This code snippet
```python
from other import x, y  # raises unused-import (W0611) for x but not for y


class MyClass:
    x = float(x)
    y = y
```
will start raising a false positive from ea13058b9fde38698515c93bf67cc6018ed0064e up until the current main branch (6505c743564452f1b298b2a7ce2507a0b280ea25).

Related: #4993.

### Command used

A false positive is only raised in combination with `--disable=all`:

```
$ python -m pylint --disable=all --enable=W0611 <path_to_file>
```
But when invoked as 
```
$ python -m pylint <path_to_file>
```
no false positive is raised, even though `unused-import (W0611)` is enabled by default.

### Pylint output

```shell
************* Module mycodesnippet
/path/to/mycodesnippet.py:1:0: W0611: Unused x imported from other (unused-import)

------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 7.50/10, +0.00)
```

### Expected behavior

No false positive should be raised.

### Pylint version

```shell
pylint 2.14.0-dev0
astroid 2.11.2
Python 3.6.15 | packaged by conda-forge | (default, Dec  3 2021, 18:49:43) 
[GCC Clang 11.1.0]
```

### OS / Environment

macOS 12.3

**Comment by DudeNr33** [comment]
While the original bug introduced by https://github.com/PyCQA/pylint/commit/ea13058b9fde38698515c93bf67cc6018ed0064e really was an issue with the ``unused-import`` not being included in the ``check_messages`` call, this is not the root cause in the current main branch.

The error is also slightly different:
When running with https://github.com/PyCQA/pylint/commit/ea13058b9fde38698515c93bf67cc6018ed0064e, the output for ``pylint --disable=all --enable=unused-import a.py`` contains error messages for both ``x`` and ``y``:
```shell
************* Module a
/Users/andreas/programming/forks/pylint/.notes/a.py:1:0: W0611: Unused x imported from other (unused-import)
/Users/andreas/programming/forks/pylint/.notes/a.py:1:0: W0611: Unused y imported from other (unused-import)
```

With the current main branch, the output only complains about ``x``:
```shell
************* Module a
a.py:1:0: W0611: Unused x imported from other (unused-import)
```

It makes no difference if we short-circuit the ``check_messages`` decorator.
The name ``x`` is not marked as consumed when only ``unused-import`` is enabled, but I have not yet figured out why.

**Comment by teake** [comment]
FWIW, enabling either `used-before-assignment (E0601)` or `undefined-variable (E0602)` together with `W0611` makes the issue go away:

```
$ python -m pylint  --disable=all --enable=E0602,W0611 <path_to_file>

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

Those are the only two rules that seem to make a difference though -- enabling any of the others does not make the issue go away.

**Comment by DudeNr33** [comment]
Thanks, this explains why in the current version the ``check_messages`` decorator does not play a role.
``used-before-assignment`` and ``undefined-variable`` translate to internal flags that are used to change the control flow in the checker class and how the consumption of names is handled.

This was introduced with https://github.com/PyCQA/pylint/commit/c5181886b59799740306829e0e21e548c50db391 which apparently already aimed at solving similar problems, but did not catch all of them. It explicitly dealt with the ``--disable=all --enable=unused-import`` configuration.
Before that commit, we get false positives for both ``x`` and ``y``.
Afterwards is the situation we have now, with only a false positive for ``x``.

**Comment by teake** [comment]
Thanks guys!
